examples/preprocessing/plot_run_ica.py

```diff
@@ -25,23 +25,45 @@
 print(__doc__)
 
 ###############################################################################
-# Fit ICA model using the FastICA algorithm, detect and inspect components
+# Read and preprocess the data. Preprocessing consists of:
+# * meg channel selection
+# * 1 - 30 Hz band-pass IIR filter
+# * epoching -0.2 to 0.5 seconds with respect to events
 
 data_path = sample.data_path()
 raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
 
 raw = mne.io.read_raw_fif(raw_fname, preload=True)
-raw.filter(1, 30, method='iir')
 raw.pick_types(meg=True, eeg=False, exclude='bads', stim=True)
+raw.filter(1, 30, method='iir')
 
 # longer + more epochs for more artifact exposure
 events = mne.find_events(raw, stim_channel='STI 014')
 epochs = mne.Epochs(raw, events, event_id=None, tmin=-0.2, tmax=0.5)
 
+###############################################################################
+# Fit ICA model using the FastICA algorithm, detect and plot components
+# explaining ECG artifacts.
+
 ica = ICA(n_components=0.95, method='fastica').fit(epochs)
 
 ecg_epochs = create_ecg_epochs(raw, tmin=-.5, tmax=.5)
 ecg_inds, scores = ica.find_bads_ecg(ecg_epochs)
 
 ica.plot_components(ecg_inds)
+
+###############################################################################
+# Plot properties of ECG components:
 ica.plot_properties(epochs, picks=ecg_inds)
+
+###############################################################################
+# Mark EOG components for removal and compare evoked response before/after:
+
+ica.exclude = ecg_inds
+ica.plot_overlay(epochs.average())
+
+###############################################################################
+# Although the epochs evoked response does not change much, we can see the
+# effects of component removal clearly on `ecg_epochs`:
+
+ica.plot_overlay(ecg_epochs.average())
```

Comment:
Yeah, that's true - this is the way it was all the time before, but without plots nobody noticed.
I added the plots because in the introduction the example says these will be performed (it says  "Subsequently, artefact detection and rejection quality are assessed. Finally, the impact on the evoked ERF is visualized." but it was never actually done in the example).
I can go half step back and only show grads but that would be dishonest. I can also experiment with the example to see whether ECG removal can go better.
On the other hand - failure to perfectly attenuate artifacts by ICA is part of life so maybe we shouldn't hide this away from the users - but instead add a commentary on why it is not perfect and how/when it could go better?